libc/src/time/ctime_s.cpp

```diff
@@ -0,0 +1,39 @@
+//===-- Implementation of ctime_s function --------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "ctime_s.h"
+#include "src/__support/CPP/limits.h"
+#include "src/__support/common.h"
+#include "src/__support/macros/config.h"
+#include "time_utils.h"
+#include <cerrno>
+
+namespace LIBC_NAMESPACE_DECL {
+
+using LIBC_NAMESPACE::time_utils::TimeConstants;
+
+LLVM_LIBC_FUNCTION(int, ctime_s,
+                   (char *buffer, size_t buffer_size, const time_t *t_ptr)) {
+  if (t_ptr == nullptr || buffer == nullptr ||
+      *time > cpp::numeric_limits<int32_t>::max()) {
+    return EINVAL;
+  }
+
+  if (buffer_size < TimeConstants::ASCTIME_MAX_BYTES) {
+    return ERANGE;
+  }
+
+  if (time_utils::asctime(time_utils::localtime(t_ptr), buffer, buffer_size) ==
+      nullptr) {
+    return EINVAL;
+  }
```

Comment:
It's llvm style to omit `{}` for single statement conditionals; please omit them.

https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements

Reply:
Actually I am aware of this but saw this pattern in many `libc` files so replicated.

Reply:
Then those places should be cleaned up, too, in a distinct PR. Please fix.

Reply:
This point still stands. Please match LLVM style here.